File: leetcode/AIcoding/3.DeepSeek-v3/python_solutions/2525-categorize-box-according-to-criteria.py
class Solution:
    def categorizeBox(self, length: int, width: int, height: int, mass: int) -> str:
        bulky = False
        heavy = mass >= 100

        volume = length * width * height
        if length >= 10**4 or width >= 10**4 or height >= 10**4 or volume >= 10**9:
            bulky = True

        if bulky and heavy:
            return "Both"
        elif bulky:
            return "Bulky"
        elif heavy:
            return "Heavy"
        else:
            return "Neither"

File: leetcode/AIcoding/2.Gemini-2.0/erlang_solutions/2525-categorize-box-according-to-criteria.erl
-module(categorize_box).
-export([categorize_box/4]).

categorize_box(Length, Width, Height, Mass) ->
  Volume = Length * Width * Height,
  Bulky = (Length >= 10000 or Width >= 10000 or Height >= 10000 or Volume >= 1000000000),
  Heavy = (Mass >= 100),
  case {Bulky, Heavy} of
    {true, true} -> "Both";
    {true, false} -> "Bulky";
    {false, true} -> "Heavy";
    {false, false} -> "Neither"
  end.